### ISSUE TITLE
CSS-препроцессоры против CSS (3.09.2018)

### ISSUE THREAD

**Issue by nicothin** [nicothin/nicothin.github.io#8]
https://nicothin.pro/page/css-preprocessors

**Comment by thrnd** [comment]
Кстати, сталкивался с мнением, что (на примере sass/scss + БЭМ) вложенность с амперсандами очень трудно читать. С БЭМ и препроцессорами у меня на данный момент опыт небольшой, но пару раз натыкался на такую трудночитабельность своего же кода. Хотя, в твоей рекомендации по использованию препроцессоров и БЭМ, ты рекомендуешь именно такой способ написания. Хотя в таком случае (если от этого способа отказаться), теряется добрая часть профита от препроцессоров. Есть ли какие-то тулзы, которые препроцессят не в css, а в, например, тот же .scss, но разворачивая все (или некоторые) вложенности с амперсандами. Чтобы писать как быстрее и удобнее, а более читабельную версию исходника складывать в хранилище исходников (в некоторых компаниях, насколько я слышал, такой подход к написанию запрещен - т.е. с амперсанд-вложенностью).

И еще =) В той самой статье ты упоминал свои собственные генераторы блоков. А почему не пользуешься инструментами Яндекса (bem create/i-bem.js)? Мне кажется, там функционал помощнее.

**Comment by nicothin** [comment]
С нечитаемостью кода с `&` не сталкивался, читаю код комфортно (и быстрее) и свой, и студентов своих.

Яндексовскими генераторами не пользуюсь, ибо полный стек БЭМа не использую.

**Comment by thrnd** [comment]
А "полный стек БЭМа" это что в таком случае?) Более сложная файловая структура (отдельные файлы для элементов и модификаторов)?

**Comment by nicothin** [comment]
Нет. Полный стек — уйма инструментов (сборщики и т.п.), позволяющие писать с bemxjst, реализующие двуэтапную шаблонизацию, уровни переопределний и пр.

**Comment by thrnd** [comment]
Понял. 

Я свой первый коммент отредактировал в момент твоего ответа:
"Есть ли какие-то тулзы, которые препроцессят не в css, а в, например, тот же .scss, но разворачивая все (или некоторые) вложенности с амперсандами. Чтобы писать как быстрее и удобнее, а более читабельную версию исходника складывать в хранилище исходников (в некоторых компаниях, насколько я слышал, такой подход к написанию запрещен - т.е. с амперсанд-вложенностью)."

**Comment by nicothin** [comment]
@randomnf

>Чтобы писать как быстрее и удобнее, а более читабельную версию исходника складывать в хранилище исходников

С амперсандом она и более читаемая.
Ну а граждан, не использующих (запрещающих) амперсанд из побуждений, что поиском по файлу целый селектор не найти, не так уж много пока, поэтому таких извращенских тулзов и нет.
